crates/ruff_linter/resources/test/fixtures/flake8_type_checking/TCH008.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+from typing import TypeAlias, TYPE_CHECKING
+
+from foo import Foo
+
+if TYPE_CHECKING:
+    from typing import Dict
+
+    OptStr: TypeAlias = str | None
+    Bar: TypeAlias = Foo[int]
+else:
+    Bar = Foo
+
+a: TypeAlias = 'int'  # TCH008
+b: TypeAlias = 'Dict'  # OK
+c: TypeAlias = 'Foo'   # TCH008
+d: TypeAlias = 'Foo[str]'  # OK
+e: TypeAlias = 'Foo.bar'  # OK
+f: TypeAlias = 'Foo | None'  # TCH008
+g: TypeAlias = 'OptStr'   # OK
+h: TypeAlias = 'Bar'   # TCH008
+i: TypeAlias = Foo['str']   # TCH008
+j: TypeAlias = 'Baz'   # OK
+k: TypeAlias = 'k | None'  # OK
+l: TypeAlias = 'int' | None  # TCH008 (because TC010 is not enabled)
+
+type B = 'Dict'  # TCH008
+type D = 'Foo[str]'  # TCH008
+type E = 'Foo.bar'  # TCH008
+type G = 'OptStr'  # TCH008
+type I = Foo['str']  # TCH008
+type J = 'Baz'  # TCH008
+type K = 'K | None'  # TCH008
+type L = 'int' | None  # TCH008 (because TC010 is not enabled)
```

Comment:
Is the fix still safe when comments are present?

example:

```python
type M = ( 'M' # hello
| None)
```

Reply:
This one should be fine as long as you don't use implicit concatenation in the annotation.

So
```
type M = ( 'M' # hello
'| None')
```
may cause issues, but should not be commonly used, since it's technically out of spec, type checkers only have to support multi-line string annotations. mypy does support this style though.

So both in this case and the TCH007 case you mentioned above this would get rid of the comment. But I suppose this explains why the fix was marked unsafe and this means TCH007 should be unsafe as well.

Reply:
It's possible to check if the annotation contains comments and use that to mark the fix as safe or unsafe:

```rs
if !checker
            .comment_ranges()
            .has_comments(annotation, checker.source())
        {
...
}
```